### ISSUE TITLE
Framework for Ray

### ISSUE THREAD

**Issue by vakker** [polyaxon/polyaxon#672]
### Feature motivation

[Ray](https://ray.readthedocs.io/en/latest/) provides a very good way to parallelize ML workloads, and also the frameworks that build on top of it ([RLlib](https://ray.readthedocs.io/en/latest/rllib.html), [RaySGD](https://ray.readthedocs.io/en/latest/raysgd/raysgd.html)) help a lot for experimentation ([Tune](https://ray.readthedocs.io/en/latest/tune.html) is also great, but Polyaxon has it's own hyperparam tuning and scheduling part).

### Feature description

I would imagine this to be a framework, similarly to the Pytorch ([this](https://github.com/polyaxon/polyaxon-examples/blob/master/in_cluster/pytorch/mnist/polyaxonfile_distributed.yml)) or Tensorflow ([this](https://github.com/polyaxon/polyaxon-examples/blob/master/in_cluster/tensorflow/cifar10/polyaxonfile_distributed.yml)) one.

### Feature implementation

I think for each experiment (or maybe experiment group) creating the Ray head and few worker nodes would be sufficient for this to work. Then the main script would communicate with the head node using its IP address.

**Comment by mouradmourafiq** [comment]
This is definitely something I would like to see supported, but currently the two integrations that are underway are spark and dask, see the [announcement](https://medium.com/polyaxon/streamlining-the-machine-learning-lifecycle-52867eeea29e).

**Comment by vakker** [comment]
Yeah, I just though it would be great at one point in the future.

**Comment by Michalos88** [comment]
Has anyone tried to actually use polyaxon (just for tracking) with ray?  @mouradmourafiq

**Comment by mouradmourafiq** [comment]
I am not sure tbh, but it should be possible to do so.

Current [tracking](https://github.com/polyaxon/polyaxon/blob/master/core/polyaxon/tracking/__init__.py) is quite rich and stores data in very readable format, with apis to consume the events as json/pandas and drive dashboards and visualizations using open source tools (notebooks, bokeh, streamlit, ...)
